drivers/fuel_gauge/bq27z746/bq27z746.h

```diff
@@ -0,0 +1,60 @@
+ /*
+  * Copyright (c) 2023, ithinx GmbH
+  * Copyright (c) 2023, Tonies GmbH
+  *
+  * SPDX-License-Identifier: Apache-2.0
+  */
+
+#ifndef ZEPHYR_DRIVERS_FUELGAUGE_BQ27Z746_GAUGE_H_
+#define ZEPHYR_DRIVERS_FUELGAUGE_BQ27Z746_GAUGE_H_
+
+#include <zephyr/drivers/i2c.h>
+
+#define BQ27Z746_MANUFACTURERACCESS    0x00
+#define BQ27Z746_ATRATE                0x02
+#define BQ27Z746_ATRATETIMETOEMPTY     0x04
+#define BQ27Z746_TEMPERATURE           0x06
+#define BQ27Z746_VOLTAGE               0x08
+#define BQ27Z746_BATTERYSTATUS         0x0A
+#define BQ27Z746_CURRENT               0x0C
+#define BQ27Z746_REMAININGCAPACITY     0x10
+#define BQ27Z746_FULLCHARGECAPACITY    0x12
+#define BQ27Z746_AVERAGECURRENT        0x14
+#define BQ27Z746_AVERAGETIMETOEMPTY    0x16
+#define BQ27Z746_AVERAGETIMETOFULL     0x18
+#define BQ27Z746_MAXLOADCURRENT        0x1E /* NOT IMPLEMENTED */
+#define BQ27Z746_MAXLOADTIMETOEMPTY    0x20 /* NOT IMPLEMENTED */
+#define BQ27Z746_AVERAGEPOWER          0x22 /* NOT IMPLEMENTED */
+#define BQ27Z746_BTPDISCHARGESET       0x24 /* NOT IMPLEMENTED */
+#define BQ27Z746_BTPCHARGESET          0x26 /* NOT IMPLEMENTED */
+#define BQ27Z746_INTERNALTEMPERATURE   0x28 /* NOT IMPLEMENTED */
+#define BQ27Z746_CYCLECOUNT            0x2A
+#define BQ27Z746_RELATIVESTATEOFCHARGE 0x2C
+#define BQ27Z746_STATEOFHEALTH         0x2E /* NOT IMPLEMENTED */
+#define BQ27Z746_CHARGINGVOLTAGE       0x30
+#define BQ27Z746_CHARGINGCURRENT       0x32
+#define BQ27Z746_TERMINATEVOLTAGE      0x34 /* NOT IMPLEMENTED */
+#define BQ27Z746_TIMESTAMPUPPER        0x36 /* NOT IMPLEMENTED */
+#define BQ27Z746_TIMESTAMPLOWER        0x38 /* NOT IMPLEMENTED */
+#define BQ27Z746_QMAXCYCLES            0x3A /* NOT IMPLEMENTED */
+#define BQ27Z746_DESIGNCAPACITY        0x3C
+#define BQ27Z746_ALTMANUFACTURERACCESS 0x3E /* NOT IMPLEMENTED */
```

Comment:
As mentioned in my comment, I think we don't need to expose alt manufacturer access through the API because it seems to be another way of just querying basic properties that can be retrieved from other fuel gauges, meaning its existence is more of a driver implementation detail.